src/stack/dora.ts

```diff
@@ -0,0 +1,165 @@
+            sdk.log('You must provide a dora team id and token for dora controller')
+            return
+        }
+
+        if (!process.env.DORA_CONTROLLER_RELEASE_TAG || !process.env.DORA_CONTROLLER_REPO) {
+            sdk.log('You must provide a dora controller release tag and repo')
+            return
+        }
+
+        if (!process.env.DORA_WRITER_RELEASE_TAG || !process.env.DORA_WRITER_REPO) {
+            sdk.log('You must provide a dora writer release tag and repo')
+            return
+        }
+
+        const doraControllerReleaseTag = process.env.DORA_CONTROLLER_RELEASE_TAG
+        const doraControllerRepo = process.env.DORA_CONTROLLER_REPO
+        const doraWriterReleaseTag = process.env.DORA_WRITER_RELEASE_TAG
+        const doraWriterRepo = process.env.DORA_WRITER_REPO
+
+        const doraRegUserName = process.env.DORA_REMOTE_REG_USERNAME
+        const doraRegPass = process.env.DORA_REMOTE_REG_PASSWORD
+        const doraAuthTeam = process.env.DORA_AUTH_TEAM
+        const doraAuthToken = process.env.DORA_AUTH_TOKEN
+
+        new KubectlProvider(this, `${this.id}-kubectl-provider`, {
+            host: this.props?.cluster?.cluster?.endpoint,
+            configPath: '/home/ops/.kube/config',
+            loadConfigFile: true,
+        })
+
+        // install dora controller
+        const ghAuthCmd = `echo ${doraRegPass} | gh auth login --with-token`
+        await Exec(ghAuthCmd)
+            .catch(err => { throw err })
+
+        const namespace = 'dora-controller-system'
+        const secretName = `${this.id}-image-secret`
+
+        const doraCtlYmlCmd = `gh release download ${doraControllerReleaseTag} -R ${doraControllerRepo} -p 'install.yaml' -O -`
+        const doraCtlYml = await this.getYaml(doraCtlYmlCmd)
+            .catch(err => { throw err })
+
+        if (doraCtlYml !== undefined) {
+            const manifests = this.setDeployImgSecret(doraCtlYml, 'dora-controller-manager', secretName)
+            this.deployManifests(manifests, 'dora-controller', namespace)
+        }
+
+        // add image pull secret to cluster
+        const imgPullSecretYmlCmd = `kubectl create secret docker-registry ${secretName} --docker-username=${doraRegUserName} --docker-password=${doraRegPass} --docker-server=ghcr.io -n ${namespace} --dry-run=client -o yaml`
+        const imgPullSecretYml = await this.getYaml(imgPullSecretYmlCmd)
+            .catch(err => { throw err })
+        if (imgPullSecretYml !== undefined) {
+            const sensitiveFields = ["data.\.dockerconfigjson"]
+            this.deployManifests(imgPullSecretYml, 'dora-controller-image-pull-secret', namespace, sensitiveFields)
+        }
+
+        // install dora metric writer
+        const doraWriterSecretYmlCmd = `kubectl create secret generic cto-client-auth --from-literal=AUTH_TEAM=${doraAuthTeam} --from-literal=AUTH_TOKEN=${doraAuthToken} -n ${namespace} --dry-run=client -o yaml`
```

Comment:
interesting way to get the manifest, works well but don't know if we want to changed for a manifest object

Reply:
Agree. That is more declarative way to define it. I feel the only reason to choose this approach was, so we are not again converting it to string.